### DISCUSSION TITLE
Cannot SSH-sign Commits in Magit

### DISCUSSION THREAD

**Discussion by ariscript** [magit/magit#5274] [Support]
# Expected Behavior

I want to be able to sign my commits using my SSH signing key. I'm able to properly sign my commits in any other method, such as the terminal inside Emacs, or any other terminal or editor.

# Obeserved  Behavior

I am unable to sign my commits using Magit, with the following error:

```
128 git … commit --
hint: Waiting for your editor to close the file...
Waiting for Emacs...
error: No private key found for public key "/Users/ari/Library/Containers/com.maxgoedjen.Secretive.SecretAgent/Data/PublicKeys/7b5ef271f9d7c095ddd0b3bda30f2693.pub"?

fatal: failed to write commit object
```

# Reproduction Steps

1. Open Emacs in a git repository
2. Make some changes
3. Open Magit and stage some files
4. Try to commit using Magit

I am on macOS, and use an app called [Secretive](https://github.com/maxgoedjen/secretive) to manage my SSH keys. Since this works in the terminal without issue (within and outside of Emacs), I don't think this should be a problem for Magit, but the error message seems to imply that it might be? Secretive offers no access to the private key, and operates as an SSH agent.

I tried to run a clean instance of Emacs to see if the issue persists but I'm not skilled enough to operate Emacs currently without the niceties of Doom.

**Comment by tarsius** [comment]
See the FAQ entries https://magit.vc/manual/magit/I-am-using-macOS-and-SOMETHING-works-in-shell-but-not-in-Magit.html and https://magit.vc/manual/magit/I-am-using-MS-Windows-and-cannot-push-with-Magit.html. Yes, one of them is about gpg and the other about windows, but the general principals still apply.
